File: golang-backend/internal/handlers/projectstatushandler/dto.go
package projectstatushandler

import "github.com/aportela/doneo/internal/handlers"

type addRequest struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	HexColor string `json:"hexColor"`
	Index    uint   `json:"index"`
}

type updateRequest struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	HexColor string `json:"hexColor"`
	Index    uint   `json:"index"`
}

type filterRequest struct {
	Name *string `json:"name"`
}

type searchRequest struct {
	Pager  handlers.PagerRequest `json:"pager"`
	Order  handlers.OrderRequest `json:"order"`
	Filter *filterRequest        `json:"filter"`
}

type ProjectStatusResponse struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	HexColor string `json:"hexColor"`
	Index    uint   `json:"index"`
}

type searchResponse struct {
	ProjectStatuses []ProjectStatusResponse `json:"projectStatuses"`
	Pager           handlers.PagerResponse  `json:"pager"`
}

File: vite-vue-ts-frontend/src/modules/task-priorities/types/dto.ts
import type { PagerQuery, PagerResult } from "../../../shared/types/pager";
import type { Order } from "../../../shared/types/order";

export type AddRequest = {
  name: string;
  hexColor: string;
  index: number;
};

export type UpdateRequest = {
  id: string;
  name: string;
  hexColor: string;
  index: number;
};

type SearchRequestFilter = {
  name?: string;
};

export type SearchRequest = {
  pager: PagerQuery;
  order: Order;
  filter?: SearchRequestFilter;
};

export type TaskPriorityResponse = {
  id: string;
  name: string;
  hexColor: string;
  index: number;
};

export type SearchResponse = {
  taskPriorities: TaskPriorityResponse[];
  pager: PagerResult;
};
